synapse/lib/coro.py

```diff
@@ -172,21 +172,26 @@ def func(*args, **kwargs):
         return GenrHelp(f(*args, **kwargs))
     return func
 
-def _exectodo(que, todo):
+def _exectodo(que, todo, logconf):
+    # This is a new process: configure logging
+    s_common.setlogging(logger, **logconf)
     func, args, kwargs = todo
     try:
-        que.put(func(*args, **kwargs))
+        ret = func(*args, **kwargs)
+        que.put(ret)
     except Exception as e:
+        logger.exception(f'Error executing spawn function {func}')
         que.put(e)
 
-async def spawn(todo, timeout=None, ctx=None):
+async def spawn(todo, timeout=None, ctx=None, log_conf=None):
     '''
     Run a todo (func, args, kwargs) tuple in a multiprocessing subprocess.
 
     Args:
         todo (tuple): A tuple of function, ``*args``, and ``**kwargs``.
         timeout (int): The timeout to wait for the todo function to finish.
         ctx (multiprocess.Context): A optional multiprocessing context object.
+        log_conf (dict): A optional set of logging configuration for the spawned process.
```

Comment:
'an' optional

Reply:
also maybe remove 'set of' from 'set of logging configuration'

Reply:
fixed